pipeline/ffmpeg.go

```diff
@@ -129,15 +129,20 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	sourceSegments := sourceManifest.GetAllSegments()
 	job.sourceSegments = len(sourceSegments)
 
-	if job.sourceSegments > 0 && job.InputFileInfo.Format != "hls" {
-		firstSeg := sourceSegments[0]
-		lastSeg := sourceSegments[job.sourceSegments-1]
-
-		if err := probeSourceSegment(transcodeRequest.RequestID, firstSeg, transcodeRequest.SourceManifestURL); err != nil {
-			return nil, err
-		}
-		if err := probeSourceSegment(transcodeRequest.RequestID, lastSeg, transcodeRequest.SourceManifestURL); err != nil {
-			return nil, err
+	if job.InputFileInfo.Format != "hls" {
+		if job.sourceSegments < 4 {
```

Comment:
@thomshutt this feels so ugly but can't think of a nicer way

Reply:
Seems fine to me!